app/browser/api/ledger.js

```diff
@@ -104,13 +102,264 @@ const fileTypes = {
+    // - notification has not already been shown yet
+    // (see https://github.com/brave/browser-laptop/issues/11021)
+    if (hasFunds(state)) {
+      const isNewInstall = state.get('firstRunTimestamp') === state.getIn(['migrations', 'btcToBatTimestamp'])
+      const hasBeenNotified = state.getIn(['migrations', 'btcToBatTimestamp']) !== state.getIn(['migrations', 'btcToBatNotifiedTimestamp'])
+      if (!isNewInstall && !hasBeenNotified) {
+        notifications.showBraveWalletUpdated()
+      }
+    }
+  },
+  onInterval: (state) => {
+    if (!getSetting(settings.PAYMENTS_NOTIFICATIONS)) {
+      return
+    }
+    if (getSetting(settings.PAYMENTS_ENABLED)) {
+      notifications.showEnabledNotifications(state)
+    } else {
+      notifications.showDisabledNotifications(state)
+    }
+  },
+  onResponse: (message, buttonIndex, activeWindow) => {
+    switch (message) {
+      case notifications.text.addFunds:
+        // See showNotificationAddFunds() for buttons.
+        // buttonIndex === 1 is "Later"; the timestamp until which to delay is set
+        // in showNotificationAddFunds() when triggering this notification.
+        if (buttonIndex === 0) {
+          appActions.changeSetting(settings.PAYMENTS_NOTIFICATIONS, false)
+        } else if (buttonIndex === 2 && activeWindow) {
+          // Add funds: Open payments panel
+          appActions.createTabRequested({
+            url: 'about:preferences#payments',
+            windowId: activeWindow.id
+          })
+        }
+        break
+
+      case notifications.text.reconciliation:
+        // buttonIndex === 1 is Dismiss
+        if (buttonIndex === 0) {
+          appActions.changeSetting(settings.PAYMENTS_NOTIFICATIONS, false)
+        } else if (buttonIndex === 2 && activeWindow) {
+          appActions.createTabRequested({
+            url: 'about:preferences#payments',
+            windowId: activeWindow.id
+          })
+        }
+        break
+
+      case notifications.text.paymentDone:
+        if (buttonIndex === 0) {
+          appActions.changeSetting(settings.PAYMENTS_NOTIFICATIONS, false)
+        }
+        break
+
+      case notifications.text.tryPayments:
+        if (buttonIndex === 1 && activeWindow) {
+          appActions.createTabRequested({
+            url: 'about:preferences#payments',
+            windowId: activeWindow.id
+          })
+        }
+        appActions.changeSetting(settings.PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED, true)
+        break
+
+      case notifications.text.walletConvertedToBat:
+        if (buttonIndex === 0) {
+          // Open backup modal
+          appActions.createTabRequested({
+            url: 'about:preferences#payments?ledgerBackupOverlayVisible',
+            windowId: activeWindow.id
+          })
+        }
+        break
+
+      default:
+        return
+    }
+
+    appActions.hideNotification(message)
```

Comment:
maybe the above default: should be just break so this always runs?

Reply:
I considered that... but since other messages may be coming through (since this fires on `ipc.on(messages.NOTIFICATION_RESPONSE`) I didn't want to change the behavior